Sources/Core/Microsoft.StreamProcessing/Aggregates/TumblingMinAggregate.cs

```diff
@@ -23,30 +23,21 @@ public TumblingMinAggregate(IComparerExpression<T> comparer)
         }
 
         public Expression<Func<MinMaxState<T>>> InitialState()
-            => () => new MinMaxState<T> { currentValue = default, currentTimestamp = InvalidSyncTime };
+            => () => new MinMaxState<T> { currentTimestamp = InvalidSyncTime };
 
         public Expression<Func<MinMaxState<T>, long, T, MinMaxState<T>>> Accumulate()
-            => (state, timestamp, input) => Accumulate(state, timestamp, input);
-
-        private MinMaxState<T> Accumulate(MinMaxState<T> state, long timestamp, T input)
-        {
-            if (state.currentTimestamp == InvalidSyncTime)
+            => (state, timestamp, input) => new MinMaxState<T>
```

Comment:
>new MinMaxState [](start = 44, length = 15)

Does this mean an extra allocation per Accumulate?

Reply:
Sadly yes. This is the one wart added (for the benefit of being able to inline the code) for the tradeoff of removing about 10 others. Let's talk more about how I might be able to get rid of this one too.